### PR TITLE
Ht 3116 namespace scrub

### PR DESCRIPTION
**Pull request by mwarin**
starting this as a draft ...

### CODE REVIEW

File: lib/scrub/max_ocn.rb [hunk 1]
```diff
@@ -45,17 +49,25 @@ def ocn
...
       @ocn = data["oclcNumber"].to_i
       f.puts @ocn
       f.close
       @ocn
     end
 
+    def make_call
+      if @mock
+        @log.info "mock call"
+        `cat spec/fixtures/max_oclc_response.json`
```
Comment (aelkiss) [inline lib/scrub/max_ocn.rb:63]: `File.read('spec/fixtures/max_oclc_response.json')`

File: lib/scrub/max_ocn.rb [hunk 2]
```diff
@@ -45,17 +49,25 @@ def ocn
...
+      else
+        `curl -s "#{OCLC_URL}"`
```
Comment (aelkiss) [inline lib/scrub/max_ocn.rb:65]: Simplest possible way of doing this in ruby that I'm aware of is:
```
require 'open-uri'

URI.open(OCLC_URL).read
```

**Comment by aelkiss** [comment]
I'm confused by the history on this branch; c361696 looks like an empty commit and 
fa997cd somehow looks like a duplicate of 
42355ba.

I'd like to see this squashed down to two commitsif possible  -- one for the autoscrub renaming and one for replacing get_max_ocn.sh.

File: lib/scrub/max_ocn.rb (hunk 2 above)
Comment (mwarin) [reply lib/scrub/max_ocn.rb:65]: yep

File: lib/scrub/max_ocn.rb (hunk 1 above)
Comment (mwarin) [reply lib/scrub/max_ocn.rb:63]: yep

**Comment by mwarin** [comment]
> I'd like to see this squashed down to two commits if possible -- one for the autoscrub renaming and one for replacing get_max_ocn.sh.

I do not see the point in this, aren't we just going to squash and merge anyways?

**Comment by aelkiss** [comment]
> > I'd like to see this squashed down to two commits if possible -- one for the autoscrub renaming and one for replacing get_max_ocn.sh.
> 
> I do not see the point in this, aren't we just going to squash and merge anyways?

I already took care of that when I fixed the duplicate & empty commits that were here. When we merge I think we'll just do 'merge pull request', not 'squash and merge', though.

File: lib/scrub/autoscrub.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    def count_file_lines
+      # zcat -f works as plain cat if @path is not in gzip format.
+      `zcat -f #{@path} | wc -l`.match(/^(\d+)/)[0].to_i
+    end
+
+  end
+end
+
+if $PROGRAM_NAME == __FILE__
```
Comment (jsteverman) [inline lib/scrub/autoscrub.rb:102]: This seems like material for bin, but it could wait or I could be convinced otherwise.

**Review by jsteverman — APPROVED** [review]

File: lib/scrub/autoscrub.rb (hunk 3 above)
Comment (mwarin) [reply lib/scrub/autoscrub.rb:102]: I'd love a proper discussion about bin vs lib for the project as a whole.

**Pull request merged**